os/root-filesystem-tricks.md

```diff
@@ -0,0 +1,71 @@
+# Root Filesystem Tricks
+This document covers setting up the root filesystem encapsulated in virtual devices like RAID arrays. Since the root filesystem must be mounted by the initramfs, special care must be taken to ensure the initramfs can find and mount it correctly. GPT partition entries have a partition type GUID that is used to specify what type of partition it is (e.g. linux filesystem). Container Linux uses special type GUIDs to indicate a partition contains the root filesystem when it is encapulated by another system, like RAID.
```

Comment:
Not thrilled with the wording of "encapsulated in virtual devices like RAID arrays".
Does the part about using the GPT partition type GUIDs belong here or is it too in depth for this doc?